torch/quantization/_equalize.py

```diff
@@ -0,0 +1,133 @@
+import torch
+import torch.nn as nn
+import copy
+
+def channel_range(input, axis=0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(input.size())
+    mins = input
+    maxs = input
+
+    # reshape input to make specified axis the last axis in input
+    temp = list(range(size_of_tensor_dim))
+    temp[size_of_tensor_dim - 1] = axis
+    temp[axis] = size_of_tensor_dim - 1
+
+    mins = mins.permute(temp)
+    maxs = maxs.permute(temp)
+
+    # minimizing over all axises except the last axis
+    for i in range(size_of_tensor_dim - 1):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
+
+    assert(mins.size()[0] == input.size()[axis])
+    return maxs - mins
+
+def scaling_channels(module1, module2, output_axis=0, input_axis=1):
```

Comment:
I think you can remove this function and move the implementation to cross_layer_equalization function